staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_work_estimator.go

```diff
@@ -175,5 +175,5 @@ func shouldListFromStorage(query url.Values, opts *metav1.ListOptions) bool {
 	// Watch cache only supports ResourceVersionMatchNotOlderThan (default).
 	unsupportedMatch := match != "" && match != metav1.ResourceVersionMatchNotOlderThan
 
-	return consistentReadFromStorage || hasContinuation || hasLimit || unsupportedMatch
+	return opts.SkipCache || consistentReadFromStorage || hasContinuation || hasLimit || unsupportedMatch
```

Comment:
this isn't enough, `isListFromCache` hard-codes a check for `watch` (which actually looks like it might currently be incorrect if it doesn't consider `resourceVersion`):

`isListFromCache := requestInfo.Verb == "watch" || !shouldListFromStorage(query, &listOptions)`